### DISCUSSION TITLE
Change the source format for matching rules (and similar)

### DISCUSSION THREAD

**Discussion by gitfish** [forcedotcom/cli#2454] [Ideas]
Is it possible to change the source format for matching rules to a single file per rule rather than a single file per object? I guess much like is done for layouts, the filename prefix would be the name of the object it applies to - then the api name for the matching rule along with the suffix - e.g.

`Lead-Some_Matching_Rule.matchingRule-meta.xml`

There is probably some limitation I'm not aware of around implementing this. I image it would be straight forward to support both legacy source format (which is basically metadata api format) and this.

There are probably source formats of a similar nature, but thought I'd raise this as an example.

**Comment by mshanemc** [comment]
Yeah, we have several open requests for flexibility in the decomposition of certain metadata types.  See the `custom registry` label https://github.com/forcedotcom/cli/discussions?discussions_q=label%3Acustom-registry

We've been batting around the idea of letting a project define a custom registry where you can augment https://github.com/forcedotcom/source-deploy-retrieve/blob/ffc5eb5127d096f5534abac05dec7dcc30b3aab5/src/registry/metadataRegistry.json#L1056 by passing in your own registry definitions.

We'd need to really spec out better what these different JSON properties do and have some code that checks that the definitions are sane.

**Reply by mshanemc** [reply]
What we don't want to do is to change this in the official registry, because that leaves people's source in an incompatible state (oh, you retrieved it on 2.7.8?  That won't work with the new change that @gitfish submitted).

custom registry would be simpler--the project's owner, if they have contributors, can communicate the change and a small number of people can deal with it.

**Reply by gitfish** [reply]
Yep, naturally you wouldn't touch it if both the legacy and new layouts couldn't be supported at the same time. @mshanemc thanks for the info around the custom registry stuff.
